src/users/userHelpers.js

```diff
@@ -37,13 +37,43 @@ const statusOrder = (presence: UserPresence): number => {
   }
 };
 
-export const sortUserList = (users: User[], presences: PresenceState): User[] =>
-  [...users].sort(
-    (x1, x2) =>
-      statusOrder(presences[x1.email]) - statusOrder(presences[x2.email])
-      || x1.full_name.toLowerCase().localeCompare(x2.full_name.toLowerCase()),
+export const sortAlphabetically = (users: User[]): User[] =>
+  [...users].sort((x1, x2) => x1.full_name.toLowerCase().localeCompare(x2.full_name.toLowerCase()));
+
+export const filterUserStartWith = (users: User[], filter: string = ''): User[] =>
+  users.filter(user => user.full_name.toLowerCase().startsWith(filter.toLowerCase()));
+
+export const filterUserEmailStartWith = (users: User[], filter: string = ''): User[] =>
+  users.filter(user => user.email.toLowerCase().startsWith(filter.toLowerCase()));
+
+export const filterUserByInitials = (users: User[], filter: string = ''): User[] =>
+  users.filter(user =>
+    user.full_name
+      .replace(/(\s|[a-z])/g, '')
+      .toLowerCase()
+      .startsWith(filter.toLowerCase()),
```

Comment:
>This will never match for a user whose full_name is, for example, "sumukha hegde".

You mean if I search for "sumukha hegde", the user **sumukha hegde** won't match ?
It actually does, because **full_name** is starting with **_filter-term_**, So it will match.

Reply:
>This will never match for a user whose full_name is, for example, "sumukha hegde".

I am not clear with this, can you explain what the issue is?

Reply:
@ray-kraesig, 
that's because we pass only the users whose name includes the filter-term,
if the filter-term is **sh**, we filter users whose name includes **sh**, which eliminates **sumukha hegde**  as it does not have sh in it.

Reply:
should it match to `sumukha hegde` if a filter-term is **shegde**?
can you explain how initial matching should work?

Reply:
user initial search is not even implemented(or not working) in the web app.
If it is not useful we can adopt second name filtering instead of initial name

Reply:
No, that's not at all what this function does, nor even what it should do. It should match by the initial letter of each name, so that `"sh"` would match `"sumukha hegde"`. 

However, as written, it won't match `"sumukha hegde"` with *any* `filter`: the first `.replace` operation will strip all spaces and lowercase letters out of the username – which, in your case, is all of them. (It does match `"Sumukha Hegde"` with `"sh"`.)

It's also pretty awful for names written in non-Roman alphabets, or even just names with diacritics. Neither _Иван Иванов_ nor _João da Silva_ are likely to be happy with the results here, and what do you even do with a name like 井&zwj;手&zwj;平&zwj;一&zwj;郎 (which might be either 井&zwj;手 平&zwj;一&zwj;郎 _Ide Heiichirō_ or 井&zwj;手&zwj;平 一&zwj;郎 _Idehira Ichirō_)?

But (as noted above) this wasn't a bug you introduced.

Reply:
... and I've therefore filed it as a separate issue, #4075. Which I probably should have done initially!

(... pun not intended.)